src/VisualStudio/TestSetup/TestExtensionManager.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Extensions;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Roslyn.VisualStudio.Test.Setup
+{
+    /// <summary>
+    /// This class causes a crash if an exception is encountered inside lightbulb extension points such as code fixes and code refactorings.
+    /// </summary>
+    /// <remarks>
+    /// This class is exported as a workspace service with layer: <see cref="ServiceLayer.Host"/>. This ensures that TestExtensionManager
+    /// is preferred over EditorLayerExtensionManager (which has layer: <see cref="ServiceLayer.Editor"/>) when running VS integration tests./>
+    /// </remarks>
+    [Shared, ExportWorkspaceServiceFactory(typeof(IExtensionManager), ServiceLayer.Host)]
+    internal class TestExtensionManager : IWorkspaceServiceFactory
+    {
+        private readonly TestExtensionErrorHandler errorHandler;
+
+        [ImportingConstructor]
+        public TestExtensionManager([Import]TestExtensionErrorHandler errorHandler)
```

Comment:
The `[Import]` attribute is unneeded here since this is already labeled as an `[ImportingConstructor]`